src/fs_core.ts

```diff
@@ -21,19 +28,57 @@ export class File {
   }
 }
 
+// shim https://developer.mozilla.org/en-US/docs/Web/API/FileSystemSyncAccessHandle
```

Comment:
```suggestion
// Shim for https://developer.mozilla.org/en-US/docs/Web/API/FileSystemSyncAccessHandle
// This is not part of the public interface.
```

Also looks like updating to typescript 5 is enough to make this available: https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1471 That should probably be a separate PR.

Reply:
Huh. Maybe there's something else you need to set because this is only available in WebWorkers? My project is on TypeScript 5.0.4 and doesn't seem to include this type. I've already used [`@types/web`](https://www.npmjs.com/package/@types/web) to get support for the WebTransport API but the editor still complains about `FileSystemSyncAccessHandle`.

Reply:
Indeed, [you need to tell Typescript](https://joshuatz.com/posts/2021/strongly-typed-service-workers/) the the file should use the `webworker` lib:

```ts
/// <reference no-default-lib="true"/>
/// <reference lib="es2020" />
/// <reference lib="webworker" />

...
```

And as soon as I did that it started complaining that the types are incompatible 🙃 
![image](https://github.com/bjorn3/browser_wasi_shim/assets/11139925/9bbe4292-665c-4167-8f33-da4fcfe0ed17)

But then adding `/// <reference lib="webworker" />` at the top of `fd_core.ts` and installing `typescript@^5.0.4` fixes that for me. I'm not adding that as a commit though because you said that'd be for a separate PR.

I've just fixed the "shim" for now because while MDN says these return a `Promise<undefined>`, which is obviously an error, I thought they meant just `undefined`. The specification and typings seem to use `void` though.